src/foam/u2/view/UnstyledTableView.js

```diff
@@ -236,8 +240,47 @@ foam.CLASS({
                       end();
                   }).
                   call(function() {
-                    if ( view.editColumnsEnabled ) return this.tag('td');
-                  })
+                    var model = view.of;
+                    var actions = model.getAxiomsByClass(foam.core.Action);
+                    var overlay = view.OverlayDropdown.create();
+                    var availableActions = actions.filter(function(action) {
+                      return action.isAvailableFor(obj);
+                    });
+
+                    // Don't show the context menu if there are no available
+                    // actions defined on the model.
+                    if ( availableActions.length === 0 ) {
+                      if ( view.editColumnsEnabled ) {
+                        return this.tag('td');
+                      } else {
+                        return;
+                      }
+                    }
+
+                    availableActions.forEach((action) => {
```

Comment:
You could call overlap.forEach(availableActions, fn) and save some code.

Reply:
And not store 'overlay' in a variable and just embed in-line below.

Reply:
> And not store 'overlay' in a variable and just embed in-line below.

The problem with that is that I need it in a variable to pass it as an argument to `positionOverlayDropdown()` and call the `open()` method on below.